### ISSUE TITLE
[v4] Error resolving variable in file: does not fall back to default value

### ISSUE THREAD

**Issue by trend-adam-kinnell** [serverless/serverless#12956]
## Description

In Serverless v4 there is an issue where failing to resolve a variable in a file (using `file(...)`) will throw an fatal error and not attempt to fall back to the defined default value.

This was not an issue in Serverless V3.

## Repro steps:

**serverless.yml**
```
# "org" ensures this Service is used with the correct Serverless Framework License Key.
org: <redacted>
# "service" is the name of this project. This will also be added to your AWS resource names.
service: test

provider:
  name: aws
  runtime: nodejs20.x

custom:
  works: ${file(./configuration.yml):this.does.exist}
  also_works: ${self:this.does.not.exist, "and it falls back to this value"}
  does_not_work: ${file(./configuration.yml):this.does.not.exist, "but it should fall back to this value"}

functions:
  hello:
    handler: handler.hello
    events:
      - httpApi:
          path: /
          method: get
```

**configuration.yml**
```
this:
  does:
    exist: true
```

We then get this error:

<img width="1310" alt="image" src="https://github.com/user-attachments/assets/4fe9e483-1f76-49d7-bd4b-37208d605d42" />

## Context 

We currently use a pattern where we reference default values in an external file `configuration.yml` to optionally provide overrides for default values based on different regions or other values.

An example might look like

```
 memorySize: ${file(./configuration.yml):${self:provider.region}.memorySize, file(./configuration.yml):default.memorySize}
```

## Environment:

Serverless Framework Version: 4.4.16

### Context

Serverless ϟ Support

This generates a report from your last Serverless Framework command (including any errors) to use for Github Issues, debugging w/ AI, or creating a support ticket w/ Serverless Inc.

Summary Report -----------------

# Service Overview
- Serverless Framework Version: 4.4.16
- Service Path: /Users/adamkinnell/r/adamtest
- Service Config File: serverless.yml
- Service Name: test
- Service Runtime: nodejs20.x
- Command: package 
- Error Code: RESOLVER_RESOLVE_VARIABLE_ERROR

## Error Message
Failed to resolve variable './configuration.yml#this.does.not.exist' with resolver 'file' and provider 'file': TypeError: Cannot read properties of undefined (reading 'exist')

## Error Stacktrace
```
at #resolveKey (file:///Users/adamkinnell/.serverless/releases/4.4.16/package/dist/sf-core.js:693:52025)
at async _ResolverManager.resolve (file:///Users/adamkinnell/.serverless/releases/4.4.16/package/dist/sf-core.js:693:57540)
at async #handlePlaceholderNode (file:///Users/adamkinnell/.serverless/releases/4.4.16/package/dist/sf-core.js:693:56251)
at async file:///Users/adamkinnell/.serverless/releases/4.4.16/package/dist/sf-core.js:693:54179
at async processNodeAndHandleCompletion (file:///Users/adamkinnell/.serverless/releases/4.4.16/package/dist/sf-core.js:693:44836)
```

## Service Config
```
# "org" ensures this Service is used with the correct Serverless Framework License Key.
org: <redacted>
# "service" is the name of this project. This will also be added to your AWS resource names.
service: test
provider:
  name: aws
  runtime: nodejs20.x
custom:
  works: ${file(./configuration.yml):this.does.exist}
  also_works: ${self:this.does.not.exist, "and it falls back to this value"}
  does_not_work: ${file(./configuration.yml):this.does.not.exist, "but it should fall back to this value"}
functions:
  hello:
    handler: handler.hello
    events:
      - httpApi:
          path: /
          method: get
```

# Service Overview
- Serverless Framework Version: 4.4.16
- Command: support --summary true

**Comment by czubocha** [comment]
Hey @trend-adam-kinnell, thank you for reporting this and providing a detailed description. The fix will be included in the next release.

**Comment by czubocha** [comment]
Hey @trend-adam-kinnell, this has been fixed and is now available in [v4.4.17](https://github.com/serverless/serverless/releases/tag/v4.4.17). Let us know if you encounter any further issues!

**Comment by trend-adam-kinnell** [comment]
Thanks so much for the quick fix! I can confirm that the issue is now resolved
